CMakeLists.txt

```
@@ -22,11 +25,22 @@ add_library(${PROJECT_NAME} STATIC)
 cvs_add_to_alias(${PROJECT_NAME})
 cvs_library_version(${PROJECT_NAME})
 
+cvs_create_version_file(NAME ${PROJECT_NAME}
+                        MAJOR ${PROJECT_VERSION_MAJOR}
+                        MINOR ${PROJECT_VERSION_MINOR}
+                        PATCH ${PROJECT_VERSION_PATCH}
+                        OUTPUT "${CMAKE_CURRENT_BINARY_DIR}/generate/cvs/common")
+
+
+generate_export_header(${PROJECT_NAME}
+  EXPORT_FILE_NAME ${CMAKE_CURRENT_BINARY_DIR}/export/cvs/common/${PROJECT_NAME_LOWER}export.hpp)
```

Если мы используем макросы экспорта, то логично было бы `-fvisibility=hidden` использовать.
```suggestion
generate_export_header(${PROJECT_NAME}
  EXPORT_FILE_NAME ${CMAKE_CURRENT_BINARY_DIR}/export/cvs/common/${PROJECT_NAME_LOWER}export.hpp)

set(CMAKE_CXX_VISIBILITY_PRESET hidden)
set(CMAKE_VISIBILITY_INLINES_HIDDEN 1)
```